build.gradle

```diff
@@ -8,62 +8,37 @@ plugins {
 	id 'idea'
 }
 
-sourceSets {
-	main {
-		java {
-			srcDir 'src/'
-		}
-	}
-	test {
-		java {
-			srcDir 'test-src/'
-		}
-		resources {
-			srcDir 'test-data/'
-		}
-	}
-
+ext {
+	teamscaleVersion = '5.9.0-rc3'
 }
 
- jar {
-	exclude ("**/CheckTest*")
- }
-
 test {
 	useJUnitPlatform()
+
 	testLogging {
 		// It's sometimes useful to see the stdout during testing
 		showStandardStreams = true
 		exceptionFormat = 'full'
 	}
 }
 
-
 repositories {
 	mavenCentral()
+
+	maven {
+		url "https://share.cqse.eu/svn/artifacts/maven"
+	}
 }
 
 dependencies {
-	implementation fileTree(dir: 'lib', exclude: '*-test-utils.jar')
+	implementation platform("com.teamscale:teamscale-bom:$teamscaleVersion")
+
+	implementation 'com.teamscale:teamscale-check-api'
+	implementation 'com.teamscale:teamscale-commons'
+	implementation 'com.teamscale:teamscale-lib-commons'

```

Comment:
Those are transitive dependencies of `teamscale-check-api` so there should be no need to specify them explicitly.

Makes it also easier from our side to refactor the packages later on

Reply:
Yes, they are transitive dependencies of `teamscale-check-api`, **but** they are **also** direct dependencies of `teamscale-custom-check-sample`, as can be seen by running `mvn dependency:analyze`, which determines that the bytecode of `teamscale-custom-check-sample` directly mentions, e.g., `org.conqat.lib.commons.collections.CollectionUtils` (from `teamscale-lib-commons`) or `eu.cqse.check.framework.scanner.ELanguage` (from `teamscale-commons`):

```
[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ custom-checks ---
Unable to process: eu.cqse.check.sample.SampleCheckUsingPhase
[WARNING] Used undeclared dependencies found:
[WARNING]    com.teamscale:teamscale-lib-commons:jar:5.9.0-rc3:compile
[WARNING]    com.teamscale:teamscale-commons:jar:5.9.0-rc3:compile
```

I thus very much prefer in making the dependencies that _do_ exist explicit.

Reply:
Ah ok, makes sense.